src/app/space/settings/areas/create-area-dialog/create-area-dialog.component.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, NO_ERRORS_SCHEMA } from '@angular/core';
+import { NgForm, NgModel } from '@angular/forms';
+import { AreaService } from 'ngx-fabric8-wit';
+import { Observable } from 'rxjs/Observable';
+import { initContext, TestContext } from 'testing/test-context';
+import { CreateAreaDialogComponent } from './create-area-dialog.component';
+
+@Component({
+  template: '<create-area-dialog></create-area-dialog>'
+})
+class HostComponent {}
+
+describe('CreateAreaDialogComponent', () => {
+  type Context = TestContext<CreateAreaDialogComponent, HostComponent>;
+  let mockAreaService: jasmine.SpyObj<AreaService>;
+
+  initContext(CreateAreaDialogComponent, HostComponent, {
+    declarations: [NgForm, NgModel],
+    imports: [],
+    providers: [
+      { provide: AreaService, useValue: mockAreaService}
+    ],
+    schemas: [NO_ERRORS_SCHEMA]
+  });
+
+  describe('#validateAreaName', () => {
+    it('should reset the current error states when called', function(this: Context) {
+      spyOn(this.testedDirective, 'resetErrors');
+      this.testedDirective.name = 'mock-name';
+      this.testedDirective.validateAreaName();
+      expect(this.testedDirective.resetErrors).toHaveBeenCalled();
+    });
+
+    it('should set the emptyNameFailure to true if the name is an empty string', function(this: Context) {
+      this.testedDirective.name = ' ';
+      this.testedDirective.validateAreaName();
+      expect(this.testedDirective.errors.uniqueValidationFailure).toBeFalsy();
+      expect(this.testedDirective.errors.exceedLengthFailure).toBeFalsy();
+      expect(this.testedDirective.errors.emptyNameFailure).toBeTruthy();
+    });
+
+    it('should set the exceedLengthFailure to true if the name is longer than 63 chars', function(this: Context) {
+      this.testedDirective.name = 'thisisanareanamethatisprettylongitsactuallymorethan63characters!';
+      this.testedDirective.validateAreaName();
+      expect(this.testedDirective.errors.emptyNameFailure).toBeFalsy();
+      expect(this.testedDirective.errors.uniqueValidationFailure).toBeFalsy();
+      expect(this.testedDirective.errors.exceedLengthFailure).toBeTruthy();
+    });
+  });
```

Comment:
Could you add tests to exercise the `uniqueValidationFailure` case, as well as to ensure that a valid name produces no errors?

Reply:
Yeah I can add some unit tests for that case.

Reply:
The commit message verb here can be `test(create-areas)`, rather than `fix(create-areas)`